backend/app/services/subscription/market_service.py

```
@@ -69,6 +71,171 @@ def _get_trigger_description(
 class SubscriptionMarketService:
     """Service class for Subscription Market operations."""
 
+    def _safe_trigger_type(self, raw_trigger_type: Any) -> SubscriptionTriggerType:
+        """Parse trigger type safely with cron fallback."""
+        try:
+            return SubscriptionTriggerType(raw_trigger_type)
+        except Exception:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)